### PR TITLE
[hotfix][streaming]Improve throwing exception for WriteSinkFunction.java

### PR DESCRIPTION
**Pull request by HuangWHWHW**
When call the `cleanFile()` method in WriteSinkFunction, it will only print "File not found" when error occurred.
And the FileNotFoundException is not only the "File not found" but also others(.e.g "Permission denied").
So I changed the print path to print exception.

For example, when a "Permission denied" happened, here is the exception information following before change:
java.lang.RuntimeException: File not found /result.txt
at java.io.FileOutputStream.open0(Native Method)
at java.io.FileOutputStream.open(FileOutputStream.java:270)
at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
at java.io.FileOutputStream.<init>(FileOutputStream.java:101)
at java.io.PrintWriter.<init>(PrintWriter.java:184)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunction.cleanFile(WriteSinkFunction.java:55)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunction.<init>(WriteSinkFunction.java:43)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunctionTest$2.<init>(WriteSinkFunctionTest.java:22)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunctionTest.testFileNotFound(WriteSinkFunctionTest.java:22)

And here is the info after change:
java.lang.RuntimeException: An error occured while cleaning the file:java.io.FileNotFoundException: /result.txt (Permission denied)
at java.io.FileOutputStream.open0(Native Method)
at java.io.FileOutputStream.open(FileOutputStream.java:270)
at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
at java.io.FileOutputStream.<init>(FileOutputStream.java:101)
at java.io.PrintWriter.<init>(PrintWriter.java:184)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunction.cleanFile(WriteSinkFunction.java:55)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunction.<init>(WriteSinkFunction.java:43)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunctionTest$2.<init>(WriteSinkFunctionTest.java:22)
at org.apache.flink.streaming.api.functions.sink.WriteSinkFunctionTest.testFileNotFound(WriteSinkFunctionTest.java:22)

### CODE REVIEW

**Comment by mbalassi** [comment]
Makes sense, will merge it.

File: flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/sink/WriteSinkFunction.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ protected void cleanFile(String path) {
 			writer.print("");
 			writer.close();
 		} catch (FileNotFoundException e) {
-			throw new RuntimeException("File not found " + path, e);
+			throw new RuntimeException("An error occurred while cleaning the file:" + e, e);
```
Comment (StephanEwen) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/sink/WriteSinkFunction.java]: Shouldn't this rather be `... file: " + e.getMessage(), e);`?

Comment (mbalassi) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/sink/WriteSinkFunction.java]: Makes sense, will change it while merging.

Comment (HuangWHWHW) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/sink/WriteSinkFunction.java]: Many thanks!
Just in another PR, you told me change the `LOG("value=" + value.toString())` to `LOG("value=" + value)`.
So maybe I had some misunderstanding before.
:-D

**Pull request closed without merging**